api/src/main/java/org/openmrs/api/impl/AdministrationServiceImpl.java

```diff
@@ -150,7 +152,68 @@ public String getGlobalProperty(String propertyName) throws APIException {
 			return null;
 		}
 		
-		return dao.getGlobalProperty(propertyName);
+		String globalProperty = dao.getGlobalProperty(propertyName);
+		if (StringUtils.isBlank(globalProperty)) {
+			return null;
+		} else {
+			GlobalProperty property = dao.getGlobalPropertyObject(propertyName);
+			if (canViewGlobalProperty(property)) {
+				return globalProperty;
+			} else {
+				throw new APIException("GlobalProperty.error.privilege.required.view", new Object[] {
+					property.getViewPrivilege().getPrivilege(), propertyName });
+			}
+		}
+	}
+	
+	private boolean canViewGlobalProperty(GlobalProperty property) {
+		if (property == null) {
```

Comment:
Are there any places where you are calling this with a potentially null global property?

Reply:
@dkayiwa no sir, did not find any place where ```canViewGlobalProperty()``` is called with potentially null global property, so I removed this null checker.